app/Voucher.php

```diff
@@ -538,22 +442,51 @@ public static function getMinMaxVoucherDates(Collection $vouchers)
     }
 
     /**
-     * Gets a set of vouchers that share a shortcode and currentstate in a sponsor
+     * Gets a set of vouchers in a range using shortcode, sponsor and voucher number.
      *
      * @param Builder $query
      * @param object $rangeDef
-     * @param string $state State if interest
      * @return Builder
      */
-    public function scopeWithRangedVouchersInState(Builder $query, $rangeDef, $state)
+    public function scopeInDefinedRange($query, $rangeDef)
     {
         return $query
-            ->where('currentstate', $state)
             ->where('code', 'REGEXP', "^{$rangeDef->shortcode}[0-9]+\$") // Just vouchers that start with our shortcode
             ->where('sponsor_id', $rangeDef->sponsor_id) // that are in the sponsor (performance, using the index)
             ->whereBetween(
                 DB::raw("cast(replace(code, '{$rangeDef->shortcode}', '') as signed)"),
                 [$rangeDef->start, $rangeDef->end]
             );
+
+    }
+
+    /**
+     * Gets a set of vouchers that are in one of given states.
+     *
+     * @param Builder $query
+     * @param array $states
+     * @return Builder
+     */
+    public function scopeInOneOfStates(Builder $query, $states)
+    {
+        return $query
+            ->whereIn('currentstate', $states)
+        ;
+    }
+
+    /**
+     * Gets a set of vouchers that are voidable.
+     *
+     * @param Builder $query
+     * @param array $states
+     * @return Builder
+     */
+    public function scopeInVoidableState(Builder $query)
+    {
+        $voidable_states = ['dispatched'];
```

Comment:
shouldn't this include `printed`, too? Maybe am not fully getting it but above - line 180 you wrote that `$voidable_states = ['printed', 'dispatched'];`

Reply:
We should only be voiding dispatched vouchers at this point